packages/cli/src/tools/__tests__/npm-test.ts

```diff
@@ -0,0 +1,91 @@
+import {getTemplateVersion} from '../npm';
+import assert from 'assert';
+
+let ref: any;
+
+global.fetch = jest.fn();
+
+function fetchReturn(json: any): void {
+  assert(global.fetch != null, 'You forgot to backup global.fetch!');
+  // @ts-ignore
+  global.fetch = jest.fn(() =>
+    Promise.resolve({json: () => Promise.resolve(json)}),
+  );
+}
+
+describe('getTemplateVersion', () => {
+  beforeEach(() => {
+    ref = global.fetch;
+  });
+  afterEach(() => {
+    global.fetch = ref;
+  });
+
+  it('should order matching versions with the most recent first', async () => {
+    const VERSION = '0.75.1';
+    fetchReturn({
+      versions: {
+        '3.2.1': {scripts: {version: VERSION}},
+        '1.0.0': {scripts: {version: '0.75.0'}},
+        '1.2.3': {scripts: {version: VERSION}},
+      },
+      time: {
+        '3.2.1': '2024-08-15T00:00:00.000Z',
+        '1.0.0': '2024-08-15T10:10:10.000Z',
+        '1.2.3': '2024-08-16T00:00:00.000Z', // Last published version
+      },
+    });
+
+    expect(await getTemplateVersion(VERSION)).toEqual('1.2.3');
+  });
+
+  it('should matching latest MAJOR.MINOR if MAJOR.MINOR.PATCH has no match', async () => {
+    fetchReturn({
+      versions: {
+        '3.2.1': {scripts: {version: '0.75.1'}},
+        '3.2.2': {scripts: {version: '0.75.2'}},
+      },
+      time: {
+        '3.2.1': '2024-08-15T00:00:00.000Z',
+        '3.2.2': '2024-08-16T00:00:00.000Z', // Last published version
+      },
+    });
+
+    expect(await getTemplateVersion('0.75.3')).toEqual('3.2.2');
+  });
+
+  it('should NOT matching when MAJOR.MINOR is not found', async () => {
+    fetchReturn({
+      versions: {
+        '3.2.1': {scripts: {version: '0.75.1'}},
```

Comment:
btw `version` in the metadata is not the best name imo, is it possible to change to some other name at this point? 🤔 e.g. `reactNativeVersion`

Reply:
No, we're committing that information every release in React Native now, and have been since the 0.74 issue.

Reply:
I'll follow up with infra changes to our deployment scripts.